core/storage/merkle/polkadot_trie_db/polkadot_trie_db.cpp

```diff
@@ -0,0 +1,477 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/merkle/polkadot_trie_db/polkadot_trie_db.hpp"
+
+#include <utility>
+
+#include "scale/byte_array_stream.hpp"
+#include "storage/merkle/polkadot_trie_db/polkadot_codec.hpp"
+#include "storage/merkle/polkadot_trie_db/polkadot_node.hpp"
+
+using kagome::common::Buffer;
+
+namespace {
+  /**
+   * Returns a subspan of a buffer
+   * See gsl::span
+   */
+  inline auto subbuffer(const Buffer &key, size_t offset = 0,
+                        size_t length = -1) {
+    return Buffer(
+        gsl::span<const uint8_t>(key.toVector()).subspan(offset, length));
```

Comment:
How does `count = -1` work here?

Reply:
It means full length in span. To the end of the range.